idaes/generic_models/properties/core/eos/enrtl.py

```diff
@@ -26,32 +27,245 @@
+                # No like species interactions
+                return Expression.Skip
+            elif ((i in molecular_set) and
+                    (j in molecular_set)):
+                return _G_appr(b, pobj, i, j, b.temperature)
+            elif (i in b.params.cation_set and j in molecular_set):
+                return sum(
+                    Y[k] * _G_appr(b, pobj, (i+", "+k), j,  b.temperature)
+                    for k in b.params.anion_set)
+            elif (j in b.params.cation_set and i in molecular_set):
+                return sum(
+                    Y[k] * _G_appr(b, pobj, (j+", "+k), i, b.temperature)
+                    for k in b.params.anion_set)
+            elif (i in b.params.anion_set and j in molecular_set):
+                return sum(
+                    Y[k] * _G_appr(b, pobj, (k+", "+i), j, b.temperature)
+                    for k in b.params.cation_set)
+            elif (j in b.params.anion_set and i in molecular_set):
+                return sum(
+                    Y[k] * _G_appr(b, pobj, (k+", "+j), i, b.temperature)
+                    for k in b.params.cation_set)
+            elif (i in b.params.cation_set and j in b.params.anion_set):
+                return sum(Y[k] * _G_appr(
+                    b, pobj, (i+", "+j), (k+", "+j), b.temperature)
+                    for k in b.params.cation_set
+                    if i != k)
+            elif (i in b.params.anion_set and j in b.params.cation_set):
+                return sum(Y[k] * _G_appr(
+                    b, pobj, (j+", "+i), (j+", "+k), b.temperature)
+                    for k in b.params.anion_set
+                    if i != k)
+            else:
+                raise BurntToast(
+                    "{} eNRTL model encountered unexpected component pair {}."
+                    .format(b.name, (i, j)))
+        b.add_component(pname+"_G",
+                        Expression(b.params.true_species_set,
+                                   b.params.true_species_set,
+                                   rule=rule_G_expr,
+                                   doc="Local interaction G term"))
+
+        # Calculate tau terms
+        def rule_tau_expr(b, i, j):
+            if ((pname, i) not in b.params.true_phase_component_set or
+                    (pname, j) not in b.params.true_phase_component_set):
+                return Expression.Skip
+            elif (i == j or
+                  (i in b.params.cation_set and j in b.params.cation_set) or
+                  (i in b.params.anion_set and j in b.params.anion_set)):
+                # No like species interactions
+                return Expression.Skip
+            elif ((i in molecular_set) and
+                    (j in molecular_set)):
+                return tau_rule(b, pobj, i, j, b.temperature)
+            else:
+                alpha = getattr(b, pname+"_alpha")
+                G = getattr(b, pname+"_G")
+                return log(G[i, j])/alpha[i, j]
```

Comment:
@andrewlee94 is the "minus" included implicitly here?

Reply:
No, that probably should be fixed.